### PR TITLE
Improve CopyAssemblies task so that it can apply to a specific platform and an specific deploydir

### PR DESCRIPTION
**Pull request by claudiamurialdo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,7 +84,15 @@ For the following steps must be executed from inside ```dotnet``` directory:
 ## How to copy assemblies to build directory?
```
Comment (anaiberta) [inline README.md:84]:
```suggestion
## How to test your changes with a GeneXus installation?
```

File: README.md [hunk 2]
```diff
@@ -84,7 +84,15 @@ For the following steps must be executed from inside ```dotnet``` directory:
...
 - ```dotnet msbuild /t:build;CopyAssemblies DotNetStandardClasses.sln```
 
-It copies the .NET assemblies to the folder build/**gxnet/bin** and .NET Core assemblies to build/**gxnetcore/bin**
+It compiles the solution and copies the .NET assemblies to the folder build/**gxnet/bin**.
```
Comment (anaiberta) [inline README.md:87]:
```suggestion
It compiles the solution and copies all the .NET assemblies to the folder build/gxnet*/bin**. Then, you can copy those files to a GeneXus installation or to your web application directory.
```

File: README.md [hunk 3]
```diff
@@ -84,7 +84,15 @@ For the following steps must be executed from inside ```dotnet``` directory:
...
+
```
Comment (anaiberta) [inline README.md:88]:
```suggestion

You can use the following parameters to customize the deploy:
- TargetFramework: only the assemblies that are generated for this framework will be deployed. Valid values are: `net462` (for GeneXus C# generator) and `net5` (for GeneXus NetCore generator).
- DeployDirectory: specifies a GeneXus installation directory.

Samples:
```

File: dotnet/Directory.Build.targets [hunk 4]
```diff
@@ -57,9 +68,9 @@
 		<Exec Command="copy $(OutDir)$(ProjectName).deps.json   $(TargetFramework)\$(ProjectName).deps.json" />
 	</Target>
 
-	<Target Name="CopyAssemblies" Condition="'$(ProjectName)'!=''">
-		<Copy SourceFiles="$(ProjectDir)/$(OutDir)$(TargetFileName)" DestinationFolder="$(MSBuildThisFileDirectory)/build/gxnetcore/bin" Condition="'$(TargetFramework)'=='netcoreapp3.1'"/>
-		<Copy SourceFiles="$(ProjectDir)/$(OutDir)$(TargetFileName)" DestinationFolder="$(MSBuildThisFileDirectory)/build/gxnet/bin" Condition="'$(TargetFramework)'!='netcoreapp3.1'"/>
+	<Target Name="CopyAssemblies" Condition="'$(ProjectName)'!='' AND Exists('$(ProjectDir)$(OutDir)')">
+		<Message Importance="high" Text="Copying $(ProjectDir)$(OutDir)$(TargetFileName) to $(DeployDirectory)$(DeployDirectoryPlatformBin)"></Message>
```
Comment (anaiberta) [inline dotnet/Directory.Build.targets:66]:
```suggestion
		<Message Importance="high" Text="Copying $(ProjectDir)\$(OutDir)$(TargetFileName) to $(DeployDirectory)$(DeployDirectoryPlatformBin)"></Message>
```

Comment (claudiamurialdo) [reply dotnet/Directory.Build.targets:66]: It is not necessary the extra separator. $(ProjectDir) ends with a directory separator. Agree?

Comment (anaiberta) [reply dotnet/Directory.Build.targets:66]: You have it in the following "copy" target; if it is not neccessary, remove it from "message" and "copy".

Comment (claudiamurialdo) [reply dotnet/Directory.Build.targets:66]: Removed at commit 344a386

File: dotnet/Directory.Build.props [hunk 5]
```diff
@@ -31,6 +31,13 @@
 		<NoWarn>NU5105;CS0618;CS8032;CS0618;SYSLIB0021;SYSLIB0023</NoWarn>
 		<IsPackable>true</IsPackable>
 	</PropertyGroup>
+	
+	<PropertyGroup Condition="!$(TargetFramework.StartsWith('net4'))">
+		<DeployDirectoryPlatformBin>\gxnetcore\bin</DeployDirectoryPlatformBin>
```
Comment (anaiberta) [inline dotnet/Directory.Build.props:36]: Don't like to have a "\" at the start; if the expected DeployDirectory is empty, assemblies are going to be copied to the root of the machine. Why don't move the "\" to the end of DeployDirectory?

**Review by anaiberta — APPROVED** [review]

File: dotnet/Directory.Build.props (hunk 5 above)
Comment (anaiberta) [reply dotnet/Directory.Build.props:36]: But i understand that is easy when defining the parameter DeployDirectory by command-line.

Comment (claudiamurialdo) [reply dotnet/Directory.Build.props]: Agree. Fixed.
<DeployDirectory> by default is $(MSBuildThisFileDirectory)build
but if the user sets it explicitly to an empty one on the command line, that happens.

**Review by anaiberta — APPROVED** [review]

**Pull request merged**